### PR TITLE
[For 2.6.x] Fix/2569  Fix generated avatars when using spaces in the public name

### LINKED ISSUE #2569: When changing public name, autogenerated avatar in sidebar is done with unstripped string
## Bug description & symptom

When user change his public name with a name beginning with "  "(space char), avatar in sidebar is made from this unstripped (no space at beginning or at ending) string so, il we put " aa" we get a, when logical behavior should be getting "aa" as string is stripped in backen so user name will be "aa" and not " aa".

empty_avatar
 
## Expected

Always create avatar with stripped (removing space at beginning and ending) string.

# Implementation

The fix trims trailing whitespaces.

### PR DESCRIPTION
**Pull request by raphj**
Related to #2569.

**For code reviewer**

- [x] Code is clear enough
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

**For developer**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] Original issue have been updated with a short resume of implemented solution

**For quality team**

- [x] Tested by quality team

### CODE REVIEW

**Comment by metzgegu** [comment]
It would be nice to have also a unit test for this case to the function `getTwoLetter()` in frontend_lib

File: functionnal_tests/cypress/integration/account/account_main_page.js [hunk 1]
```diff
@@ -8,9 +8,17 @@ describe('Account page', function () {
...
     cy.visit('/')
   })
-  it("should be able it access it through header's button", function () {
+  it("should be able it access it through header's button", () => {
     cy.get('[data-cy=menuprofil__dropdown__button]').click()
     cy.get('[data-cy=menuprofil__dropdown__account__link]').click()
     cy.get('.userinfo')
   })
+
+  it("Avatar should not be empty", () => {
```
Comment (Skylsmoi) [inline functionnal_tests/cypress/integration/account/account_main_page.js:17]: as from standard.js, strings must be single quoted
Also, the test seems to be about untrimmed strings, the test name doesn't reflect it

**Comment by Skylsmoi** [comment]
> It would be nice to have also a unit test for this case to the function `getTwoLetter()` in frontend_lib

I agree, if possible, prefer unit tests over cypress test. It is easier to write, easier to maintain and more robust.

File: functionnal_tests/cypress/integration/account/account_main_page.js [hunk 2]
```diff
@@ -6,11 +9,39 @@ describe('Account page', function () {
...
     cy.loginAs('users')
-    cy.visit('/')
+    cy.visitPage({ pageName: PAGES.HOME })
   })
-  it("should be able it access it through header's button", function () {
-    cy.get('[data-cy=menuprofil__dropdown__button]').click()
-    cy.get('[data-cy=menuprofil__dropdown__account__link]').click()
-    cy.get('.userinfo')
+  describe("header's buttons", () => {
+    it("should be able it access it through header's buttons", () => {
```
Comment (Skylsmoi) [inline functionnal_tests/cypress/integration/account/account_main_page.js:15]: I think there are some mistakes in this test name

File: functionnal_tests/cypress/integration/account/account_main_page.js [hunk 3]
```diff
@@ -6,11 +9,39 @@ describe('Account page', function () {
...
+
+      cy.get('[data-cy=menuprofil__dropdown__account__link]')
+        .click()
+
+      cy.get('.userinfo')
+    })
+  })
+
+  describe('Avatar', () => {
+    it('Avatar letter should be displayed correctly when the profile name has a name with spaces', () => {
```
Comment (Skylsmoi) [inline functionnal_tests/cypress/integration/account/account_main_page.js:28]:
```suggestion
    it('Avatar letters should be displayed correctly when the profile name has spaces', () => {
```

**Comment by gbonaspetti** [comment]
Tested and verified on branch.

**Pull request merged**